pkg/github/github_secrets.go

```diff
@@ -0,0 +1,144 @@
+package github
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-github/v53/github"
+	"golang.org/x/crypto/nacl/box"
+	"golang.org/x/oauth2"
+)
+
+func encodeWithPublicKey(text string, publicKey string) (string, error) {
+	publicKeyBytes, err := base64.StdEncoding.DecodeString(publicKey)
+	if err != nil {
+		return "", err
+	}
+
+	var publicKeyDecoded [32]byte
+	copy(publicKeyDecoded[:], publicKeyBytes)
+
+	encrypted, err := box.SealAnonymous(nil, []byte(text), (*[32]byte)(publicKeyBytes), rand.Reader)
+	if err != nil {
+		return "", err
+	}
+
+	encryptedBase64 := base64.StdEncoding.EncodeToString(encrypted)
+
+	return encryptedBase64, nil
+}
+
+func createRepositorySecret(ctx context.Context, privateURL, owner, repo, secretName, secretValue, authToken string) error {
+	client, err := newGithubClientWithToken(authToken, privateURL)
+	if err != nil {
+		return err
+	}
+
+	var key *github.PublicKey
+	key, _, err = client.Actions.GetRepoPublicKey(ctx, owner, repo)
+	if err != nil {
+		return err
+	}
+
+	encryptedSecret, err := encodeWithPublicKey(secretValue, *key.Key)
+	if err != nil {
+		return err
+	}
+
+	secret := &github.EncryptedSecret{
+		Name:           secretName,
+		EncryptedValue: encryptedSecret,
+	}
+
+	_, err = client.Actions.CreateOrUpdateRepoSecret(ctx, owner, repo, secret)
+	return err
+}
+
+func createRepositoryEnvSecret(ctx context.Context, privateURL, owner, repo, env, secretName, secretValue, authToken string) error {
```

Comment:
These are a lot of parameters! Suggest creating a small struct(s) to hold these, and then pass that/those instead.

Maybe even having a dedicated `Client` type so you don't have to pass `privateURL` and `authToken` everywhere.

Such as:

```go
type Client struct {
	PrivateURL string
	AuthToken  string
}

type Repository struct {
	Owner string
	Name  string
}

type Environment struct {
	Name       string
	Repository Repo
}

type Secret struct {
	Name  string
	Value string
}

func (c *Client) CreateRepositoryEnvSecret(ctx context.Context, env Environment, secret Secret) error {
}
```

Reply:
I decided to use the GithubSpec struct instead.
It was already there and I keep the changes consistent if that also changes.